### PR TITLE
updated node service

### PR DESCRIPTION
**Pull request by abildw**
updated 1.1 controller and node service in preparation for Swagger 2.0 Node Controller.
Still passes all unit testing.

@geoff-reid @dalebremner @brianparry @srinia6 @RackHD/corecommitters

### CODE REVIEW

File: lib/api/1.1/northbound/nodes.js [hunk 1]
```diff
@@ -430,12 +318,7 @@ function nodesRouterFactory (
      */
 
     router.post('/nodes/:macaddress/dhcp/whitelist', rest(function(req) {
```
Comment (benbp) [inline lib/api/1.1/northbound/nodes.js]: These whitelist routes are actually just dead routes if you feel so inclined to delete them.

**Comment by benbp** [comment]
:+1:

**Comment by benbp** [comment]
As a general trend, I think we should move the unit test code for services out of the api tests and into the services tests directory when we move logic out of the api and into services, but it's a mix all over the place right now so it's not like I think this PR should be blocked or anything (just something to keep in mind). Good work!

**Comment by heckj** [comment]
:+1:

**Pull request merged**